micro_manager/snapshot/snapshot.py

```diff
@@ -144,7 +143,10 @@ def solve(self) -> None:
                     list_of_output_files,
                     self._parameter_space_size,
                 )
-        self._logger.info("Snapshot computation completed.")
+        else:
+            self._data_storage.set_status(self._output_file_path, "finished")
+        if self._rank == 0:
+            self._logger.info("Snapshot computation completed.")
```

Comment:
Is it possible that rank 0 finishes the snapshot computation before other ranks? If yes, there should be a MPI Wait command before the logging.

Reply:
It shouldn't because rank 0 merges the output files from all ranks after completion in a parallel run. The completion statement is written after rank 0 merges the results.